File: src/mobile/android/intelibill_mobile/lib/src/features/inventory/data/dto/item_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'item_dto.freezed.dart';
part 'item_dto.g.dart';

@freezed
sealed class ItemDto with _$ItemDto {
  const factory ItemDto({
    @JsonKey(name: 'id') required String id,
    @JsonKey(name: 'name') required String name,
    @JsonKey(name: 'barcode') required String barcode,
    @JsonKey(name: 'description') String? description,
    @JsonKey(name: 'uom') required String uom,
    @JsonKey(name: 'isActive') required bool isActive,
    @JsonKey(name: 'currentStock') @Default(0.0) double currentStock,
  }) = _ItemDto;

  factory ItemDto.fromJson(Map<String, dynamic> json) =>
      _$ItemDtoFromJson(json);
}

File: src/backend/Intelibill.Application/Features/Items/DTOs/ItemDto.cs
namespace Intelibill.Application.Features.Items.DTOs;

public sealed record ItemDto(
    Guid Id,
    string Name,
    string Barcode,
    string? Description,
    string Uom,
    bool IsActive,
    decimal CurrentStock,
    decimal UnitPrice,
    decimal CurrentStockValue,
    decimal ReorderLevel,
    string StockStatus,
    string? HsnCode,
    decimal DefaultTaxRatePercent,
    bool DefaultTaxIncluded);

public sealed record ItemCatalogResultDto(
    IReadOnlyList<ItemDto> Items,
    int TotalCount,
    int PageNumber,
    int PageSize,
    ItemCatalogSummaryDto Summary);

public sealed record ItemCatalogSummaryDto(
    int TotalItems,
    int ActiveItems,
    int InactiveItems,
    int RunningLowStockCount,
    int CriticalStockCount,
    decimal TotalStockValue);
